server/rest/src/main/java/org/infinispan/rest/stream/CacheEntryStreamProcessor.java

```diff
@@ -112,28 +116,51 @@ private static byte[] metadataBytes(CacheEntry<?, ?> currentEntry) {
       if (!(currentEntry instanceof InternalCacheEntry)) {
          return null;
       }
+
+      Metadata metadata = currentEntry.getMetadata();
       InternalCacheEntry ice = (InternalCacheEntry) currentEntry;
       // add metadata
-      long lifespanInSeconds = ice.getLifespan();
+      long lifespanInSeconds = metadata.lifespan();
       if (lifespanInSeconds > -1) {
          lifespanInSeconds = TimeUnit.MILLISECONDS.toSeconds(lifespanInSeconds);
       }
-      long maxIdleInSeconds = ice.getMaxIdle();
+      long maxIdleInSeconds = metadata.maxIdle();
       if (maxIdleInSeconds > -1) {
          maxIdleInSeconds = TimeUnit.MILLISECONDS.toSeconds(maxIdleInSeconds);
       }
       long created = ice.getCreated();
       long lastUsed = ice.getLastUsed();
       long expiryTime = ice.getExpiryTime();
-      return metadataBytes(lifespanInSeconds, maxIdleInSeconds, created, lastUsed, expiryTime);
+
+      EntryVersion version = metadata.version();
+      return metadataBytes(lifespanInSeconds, maxIdleInSeconds, created, lastUsed, expiryTime, version);
    }
 
-   private static byte[] metadataBytes(long timeToLiveSeconds, long maxIdleTimeSeconds, long created, long lastUsed, long expireTime) {
-      return ("\"timeToLiveSeconds\":" + timeToLiveSeconds + ","
-            + "\"maxIdleTimeSeconds\":" + maxIdleTimeSeconds + ","
-            + "\"created\":" + created + ","
-            + "\"lastUsed\":" + lastUsed + ","
-            + "\"expireTime\":" + expireTime)
-            .getBytes();
+   private static byte[] metadataBytes(long timeToLiveSeconds, long maxIdleTimeSeconds, long created, long lastUsed,
+                                       long expireTime, EntryVersion entryVersion) {
+      Long version = null;
+      Integer topologyId = null;
+
+      if (entryVersion instanceof NumericVersion) {
+         version = ((NumericVersion) entryVersion).getVersion();
+      } else if (entryVersion instanceof SimpleClusteredVersion) {
+         version = ((SimpleClusteredVersion) entryVersion).getVersion();
+         topologyId = ((SimpleClusteredVersion) entryVersion).getTopologyId();
+      }
+      Json metadata = Json.object("timeToLiveSeconds", timeToLiveSeconds,
```

Comment:
Using Json here seems like overkill.

Can't we just change it to use a StringBuilder and then do `sb.append(",\"version\":).append(version);` and then call `sb.toString().getBytes();` at the end?

Reply:
agree

Reply:
In fact, I was in doubt if using a string builder and I agree with you both. I'm changing it. Thanks